src/contracts/core/Slasher.sol

```diff
@@ -9,608 +9,94 @@ import "../permissions/Pausable.sol";
 import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
 import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
 
+
 /**
- * @title The primary 'slashing' contract for EigenLayer.
- * @author Layr Labs, Inc.
- * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
- * @notice This contract specifies details on slashing. The functionalities are:
- * - adding contracts who have permission to perform slashing,
- * - revoking permission for slashing from specified contracts,
- * - tracking historic stake updates to ensure that withdrawals can only be completed once no middlewares have slashing rights
- * over the funds being withdrawn
- */
+ * @notice This contract is not in use as of the Eigenlayer M2 release.
+ *
+ * Although many contracts reference it as an immutable variable, they do not
+ * interact with it and it is effectively dead code. The Slasher was originally
+ * deployed during Eigenlayer M1, but remained paused and unused for the duration
+ * of that release as well.
+ *
+ * Eventually, slashing design will be finalized and the Slasher will be finished
+ * and more fully incorporated into the core contracts. For now, you can ignore this
+ * file. If you really want to see what the deployed M1 version looks like, check
+ * out the `init-mainnet-deployment` branch under "releases."
+ *
+ * This contract is a stub that maintains its original interface for use in testing
+ * and deploy scripts. Otherwise, it does nothing.
+ */ 
 contract Slasher is Initializable, OwnableUpgradeable, ISlasher, Pausable {
-    using StructuredLinkedList for StructuredLinkedList.List;
-
-    uint256 private constant HEAD = 0;
-
-    uint8 internal constant PAUSED_OPT_INTO_SLASHING = 0;
-    uint8 internal constant PAUSED_FIRST_STAKE_UPDATE = 1;
-    uint8 internal constant PAUSED_NEW_FREEZING = 2;
-
-    /// @notice The central StrategyManager contract of EigenLayer
-    IStrategyManager public immutable strategyManager;
-    /// @notice The DelegationManager contract of EigenLayer
-    IDelegationManager public immutable delegation;
-    // operator => whitelisted contract with slashing permissions => (the time before which the contract is allowed to slash the user, block it was last updated)
-    mapping(address => mapping(address => MiddlewareDetails)) internal _whitelistedContractDetails;
-    // staker => if their funds are 'frozen' and potentially subject to slashing or not
```

Comment:
I think we're leaving the current Slasher as-is. It isn't called by anything, so we don't need to change it at all.

This change is primarily to get it out of the repo so that auditors/bounty hunters don't go poking around at something that doesn't exist.

Reply:
Ah okay, so not upgrading - SGTM.